recipes/folly/all/conanfile.py

```diff
@@ -126,14 +126,47 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.filenames["cmake_find_package"] = "folly"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "folly"
+        self.cpp_info.names["cmake_find_package"] = "Folly"
+        self.cpp_info.names["cmake_find_package_multi"] = "Folly"
+        self.cpp_info.names["pkg_config"] = "libfolly"
+        self.cpp_info.components["libfolly"].names["cmake_find_package"] = "folly"
+        self.cpp_info.components["libfolly"].names["cmake_find_package_multi"] = "folly"
+        self.cpp_info.components["libfolly"].names["pkg_config"] = "libfolly"
+        self.cpp_info.components["libfolly"].libs = tools.collect_libs(self)
+        self.cpp_info.components["libfolly"].requires = [
+            "boost::boost",
+            "bzip2::bzip2",
+            "double-conversion::double-conversion",
+            "gflags::gflags",
+            "glog::glog",
+            "libevent::libevent",
+            "lz4::lz4",
+            "openssl::openssl",
+            "snappy::snappy",
+            "zlib::zlib",
+            "zstd::zstd"
+        ]
+        if Version(self.version) >= "2019.01.01.00":
+            self.cpp_info.components["libfolly"].requires.extend([
+                "libdwarf::libdwarf",
+                "libsodium::libsodium",
+                "xz_utils::xz_utils"
+            ])
+            if self.settings.os == "Linux":
+                self.cpp_info.components["libfolly"].requires.extend([
+                    "libiberty::libiberty",
+                    "libunwind::libunwind"
+                ])
+        if Version(self.version) >= "2020.08.10.00":
+            self.cpp_info.components["libfolly"].requires.append("fmt::fmt")
         if self.settings.os == "Linux":
-            self.cpp_info.system_libs.extend(["pthread", "dl"])
+            self.cpp_info.components["libfolly"].system_libs.extend(["pthread", "dl"])
         elif self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
-            self.cpp_info.system_libs.extend(["ws2_32", "Iphlpapi", "Crypt32"])
+            self.cpp_info.components["libfolly"].system_libs.extend(["ws2_32", "Iphlpapi", "Crypt32"])
```

Comment:
Shouldn't it be `elif self.settings.os == "Windows"` only?

Reply:
does migwn use the same system libs?

Reply:
sure